weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/traceServerClient.ts

```diff
@@ -108,3 +108,89 @@ export class TraceServerClient {
     return res;
   };
 }
+
+const MAX_CHUNK_SIZE = 10000;
+
+/**
+ * Use this function to query calls from the trace server. This function will
+ * handle chunking the results if the user requests more than `MAX_CHUNK_SIZE`
+ * calls. This is to protect the server from returning too much data at once.
+ *
+ * Note: we should probably roll this into the `callsQuery` directly, but I don't
+ * want to change the API right now to support cancelation.
+ */
+export const chunkedCallsQuery = (
```

Comment:
i like that we pulled this out, makes useCalls much easier to understand

Reply:
👍